magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL2_to_DL3.py

```diff
@@ -0,0 +1,314 @@
+    DL3_config["source_ra"] = f"{ra} deg"
+    DL3_config["source_dec"] = f"{dec} deg"
+    conf = {
+        "mc_tel_ids": config_dict["mc_tel_ids"],
+        "dl2_to_dl3": DL3_config,
+    }
+
+    conf_dir = f"{target_dir}/v{__version__}/{source_name}"
+    os.makedirs(conf_dir, exist_ok=True)
+
+    file_name = f"{conf_dir}/config_DL3.yaml"
+
+    with open(file_name, "w") as f:
+
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def DL2_to_DL3(
+    target_dir,
+    source,
+    env_name,
+    IRF_dir,
+    df_LST,
+    cluster,
+    MC_v,
+    version,
+    nice,
+    IRF_theta_cuts_type,
+    LST_date,
+):
+    """
+    This function creates the bash scripts to run lst1_magic_dl2_to_dl3.py on the real data.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    source : str
+        Source name
+    env_name : str
+        Conda enviroment name
+    IRF_dir : str
+        Path to the IRFs
+    df_LST : :class:`pandas.DataFrame`
+        Dataframe collecting the LST1 runs (produced by the create_LST_table script)
+    cluster : str
+        Cluster system
+    MC_v : str
+        Version of MC processing
+    version : str
+        Version of the input (stereo subruns) data
+    nice : int or None
+        Job priority
+    IRF_theta_cuts_type : str
+        Type of IRFS (global/dynamic)
+    LST_date : list
+        List of the dates to be processed (from list_from_h5)
+    """
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+
+    # Loop over all nights
+    outdir = f"{target_dir}/v{__version__}/{source}/DL3/logs"
+    File_list = np.sort(glob.glob(f"{outdir}/ST*.txt"))
+    for file in File_list:
+        night = file.split("_")[-1].replace(".txt", "")
+        if str(night) not in LST_date:
+            continue
+        with open(file, "r") as f:
+            runs = f.readlines()
+            process_size = len(runs) - 1
+            run_new = []
+            for run in runs:
+                single_run_new = (
+                    "/".join(run.split("/")[:-6])
+                    + f"/v{version}/"
+                    + run.split("/")[-5]
+                    + "/DL2/"
+                    + run.split("/")[-2]
+                    + "/"
+                    + run.split("/")[-1].replace("dl1_stereo", "dl2")
+                )
+                run_new.append(single_run_new)
+            with open(file, "w") as g:
+                g.writelines(run_new)
+
+        nsb = file.split("/")[-1].split("_")[1]
+        period = file.split("/")[-1].split("_")[0]
+        dec = df_LST[df_LST.source == source].iloc[0]["MC_dec"]
+        if np.isnan(dec):
+            print(f"MC_dec is NaN for {source}")
+            continue
+        dec = str(dec).replace(".", "")
+        dec = str(dec).replace("-", "min_")
+        if IRF_theta_cuts_type == "global":
```

Comment:
I think it is more versatile if we pass e.g. "g_dyn_0.9_th_glo_0.2" directly as the cuts type